File: apps/web/src/lib/types/ai.ts
export interface ChallengeExtractionResponse {
  suggested_title: string;
  suggested_summary: string;
  suggested_domain?: string | null;
  affected_group_notes?: string | null;
  frequency_notes?: string | null;
  duration_notes?: string | null;
  current_situation_notes?: string | null;
  missing_information: string[];
  source_language: string;
}

export interface QualificationSuggestionResponse {
  suggested_route: "SERVICE" | "CLARIFY" | "RESEARCH_REVIEW" | "INNOVATION_CHALLENGE";
  reasoning_summary: string;
  evidence_considered: string[];
  missing_information: string[];
  limitations: string;
}

export interface DuplicateCandidate {
  challenge_id: string;
  title: string;
  explanation: string;
}

export interface DuplicateSuggestionResponse {
  possible_duplicate_challenge_ids: string[];
  duplicate_candidates: DuplicateCandidate[];
  reasoning_summary: string;
}

export interface HEICandidateSuggestion {
  organization_id: string;
  organization_name?: string | null;
  relevant_capabilities: string[];
  relevance_explanation: string;
  limitations?: string | null;
}

export interface HEICandidateSuggestionResponse {
  suggested_candidates: HEICandidateSuggestion[];
  reasoning_summary: string;
  limitations: string;
}

export interface EvidenceSummaryResponse {
  summary: string;
  source_evidence_ids: string[];
  missing_evidence: string[];
  uncertainties: string[];
}

File: apps/api/app/schemas/ai.py
import uuid
from typing import Any, Dict, List, Optional
from pydantic import BaseModel, ConfigDict, Field

from app.core.enums import QualificationRoute


class AIResponseBase(BaseModel):
    is_available: bool = True
    message: Optional[str] = None
    disclaimer: str = Field(
        default="AI advisory suggestion only. Non-authoritative; human decision remains final."
    )


class ChallengeExtractionRequest(BaseModel):
    raw_text: str = Field(..., min_length=1, max_length=10000)


class ChallengeExtractionResponse(AIResponseBase):
    suggested_title: str = Field(default="")
    suggested_summary: str = Field(default="")
    suggested_domain: str = Field(default="GENERAL")
    affected_group_notes: Optional[str] = None
    frequency_notes: Optional[str] = None
    duration_notes: Optional[str] = None
    current_situation_notes: Optional[str] = None
    missing_information: List[str] = Field(default_factory=list)
    source_language: str = Field(default="English / Hinglish")


class QualificationSuggestionRequest(BaseModel):
    challenge_id: uuid.UUID


class QualificationSuggestionResponse(AIResponseBase):
    suggested_route: Optional[QualificationRoute] = None
    reasoning_summary: str = Field(default="")
    evidence_considered: List[str] = Field(default_factory=list)
    missing_information: List[str] = Field(default_factory=list)
    limitations: str = Field(
        default="AI suggestion based on current challenge text and evidence metadata."
    )


class DuplicateCandidateItem(BaseModel):
    challenge_id: uuid.UUID
    title: str
    similarity_reason: str


class DuplicateSuggestionResponse(AIResponseBase):
    possible_duplicates: List[DuplicateCandidateItem] = Field(default_factory=list)


class HEICandidateItem(BaseModel):
    organization_id: uuid.UUID
    organization_name: str
    relevant_capabilities: List[str] = Field(default_factory=list)
    relevance_explanation: str


class HEICandidateSuggestionResponse(AIResponseBase):
    suggested_candidates: List[HEICandidateItem] = Field(default_factory=list)
    limitations: str = Field(
        default="Suggestions derived strictly from registered HEI active capabilities."
    )


class EvidenceSummaryResponse(AIResponseBase):
    summary_text: str = Field(default="")
    source_evidence_ids_used: List[str] = Field(default_factory=list)
    missing_evidence: List[str] = Field(default_factory=list)
    uncertainties: List[str] = Field(default_factory=list)
